util/tune/current_settings.rb

```diff
@@ -211,62 +215,44 @@ def puppetdb_java_args
 #
 # @example
 #   value = get_conf_parameter(file, parameter)
-#   value = get_conf_parameter(file, parameter, /(example_a|example_b)/)
 #
-# TODO: update to eliminate the use of exclusions
-#   See https://tickets.puppetlabs.com/browse/SLV-531
-#
-def get_conf_parameter(file, parameter, exclusions = DEFAULT_EXCLUSION)
-  value = if File.exist? file
-
-            # this pattern extracts the value from the result with the '\K'
-            match_pattern = /#{parameter} = \K[^\s]+/
-
-            # eliminate comments, search for the parameter, apply exclusions, extract the value
-            File.open(file).grep_v(/^#/).grep(/#{parameter}/).grep_v(exclusions)[0].match(match_pattern)
+def get_conf_parameter(file, parameter)
+  return NA unless File.exist? file
 
-          else
-            NA
-          end
-  value
+  conf = File.read(file)
+  match_pattern = /^\s*#{parameter} = (\S+)/
+  conf_scan = conf.scan(match_pattern)
+  conf_scan.last ? conf_scan.last[0] : NA
```

Comment:
:pick: could you change the index number to `first` to prevent the mixing of `first/last` and index references?

Reply:
updated